packages/diagnostic/src/disgnostic.ts

```diff
@@ -0,0 +1,84 @@
+import ts from 'typescript';
+import { ESLint } from 'eslint';
+import Debugger from 'debug';
+
+const debug = Debugger('rehearsal:diagnostic');
+
+export async function diagnose(srcDir: string): Promise<void> {
+  try {
+    const program = compileProgram(srcDir);
+    const diagnostics = getDiagnostics(program);
+    diagnostics.forEach((diagnostic) => {
+      console.log(diagnostic.messageText);
+
+      if (diagnostic.file) {
+        lint(diagnostic.file);
+      }
+    });
+  } catch (e) {
+    debug('Diagnostic failed', e);
+  }
+}
+
+function compileProgram(srcDir: string): ts.Program {
+  const configFile = ts.findConfigFile(
+    srcDir,
+    ts.sys.fileExists,
+    'tsconfig.json'
```

Comment:
We may need to have some sort of configuration that allows us to find the correct `tsconfig.json` since people may have different setups for different environments.

Reply:
Will add options in the the next PR